gui/default/syncthing/core/syncthingController.js

```diff
@@ -1586,11 +1607,17 @@ angular.module('syncthing.core')
 
         $scope.saveIgnores = function () {
             if (!$scope.editingExisting) {
+                $scope.newIgnores.edited = true;
+                $scope.newIgnores.ignores = $('#editIgnores textarea').val().split('\n')
                 return;
```

Comment:
I'd say it's separation of concerns. It's one function that does two things based on some magic flag. Also, saveIgnores that actually doesn't save anything in this case (just stores them somewhere) sounds fishy.